zipkin4net/Criteo.Profiling.Tracing/Utils/TraceHelper.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Criteo.Profiling.Tracing.Utils
+{
+    public static class TraceHelper
+    {
+        /// <summary>
+        /// Runs the action and adds an error annotation in case of failure
+        /// </summary>
+        /// <param name="action"></param>
+        public static void TracedAction(Action action)
+        {
+            try
+            {
+                action();
+            }
+            catch (Exception ex)
+            {
+                Trace.Current?.Record(Annotations.Tag("error", ex.Message));
```

Comment:
This shouldn't end up being a problem...the `?.` operator is a feature of the complier/Roslyn and came with VS2015+ rather than being tied to a specific version of the CLR.  I can switch it to `if (x != null) {...}` if you'd like though.